src/plugins/discover/public/components/discover_grid/discover_grid_columns.tsx

```diff
@@ -76,7 +76,10 @@ function buildEuiGridColumn({
   dataView: DataView;
   defaultColumns: boolean;
   isSortEnabled: boolean;
-  services: DiscoverServices;
+  services: {
+    toastNotifications: DiscoverServices['toastNotifications'];
+    hasEditDataViewPermission: () => boolean;
```

Comment:
I would suggest to don't mix services with other custom props like `hasEditDataViewPermission` and pass the whole `dataViewFieldEditor` instead for consistency or move `hasEditDataViewPermission` outside of `services`.